css/autotheme.css

```diff
@@ -0,0 +1,68 @@
+.autotheme {
+  display: flex;
+  flex-direction: column;
+  align-items: stretch;
+}
+
+.autotheme a {
+  display: flex;
+  flex: auto;
+  align-items: center;
+
+  color: var(--text-color);
+  font-style: normal;
+}
+
+.autotheme i {
+  width: unset;
+  height: 21px; /* same as lineheight */
+  margin-left: 1em;
+  align-self: center;
+}
+
+.autotheme-palette {
+  display: flex;
+  justify-content: space-around;
+
+  flex: auto;
+}
+
+body:not(.has-autotheme) .hide-when-no-autotheme {
+  display: none;
+}
+
+.palette-item {
+  flex: 0 1 50px;
+  min-height: 0;
+  min-width: 0;
+  border: 1px solid #ddd;
+
+  color: transparent;
+}
+
+.palette-item:before{
+  content: "";
+
+  display: block;
+  padding-top: 100%;
```

Comment:
this is a trick to make it square. Comes from http://www.mademyday.de/css-height-equals-width-with-pure-css.html